src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+            const auto end_coord = coords.back();
+
+            const auto first = std::begin(coords) + 1;
+            const auto last = std::end(coords) - 1;
+
+            auto snuggles = false;
+
+            using namespace util::coordinate_calculation;
+
+            // In addition, if it's a right/left turn we expect the rightmost/leftmost
+            // turn at `c` to be more or less ~90 degree for a Sliproad scenario.
+            double deviation_from_straight = 0;
+
+            if (is_right_turn)
+            {
+                snuggles = std::all_of(first, last, [=](auto each) { //
+                    return !isCCW(start_coord, each, end_coord);
+                });
+
+                const auto rightmost = next->intersection.getRightmostRoad();
+                deviation_from_straight = angularDeviation(rightmost.angle, STRAIGHT_ANGLE);
+            }
+            else if (is_left_turn)
+            {
+                snuggles = std::all_of(first, last, [=](auto each) { //
+                    return isCCW(start_coord, each, end_coord);
+                });
+
+                const auto leftmost = next->intersection.getLeftmostRoad();
+                deviation_from_straight = angularDeviation(leftmost.angle, STRAIGHT_ANGLE);
+            }
+
+            // The data modelling for small Sliproads is not reliable enough.
+            // Only check for curvature and ~90 degree when it makes sense to do so.
+            const constexpr auto MIN_LENGTH = 3.;
+
+            const auto length = haversineDistance(node_info_list[intersection_node_id], //
+                                                  node_info_list[next->node]);
+
+            const double perpendicular_angle = 90 + FUZZY_ANGLE_DIFFERENCE;
+
+            if (length >= MIN_LENGTH)
+            {
+                if (!snuggles)
+                    continue;
+
+                if (deviation_from_straight > perpendicular_angle)
+                    continue;
+            }
+        }
+
+        // Check for area under triangle `bdc`.
+        //
+        // a ... b .... c .
+        //       `      .
+        //         `    .
+        //           `  .
+        //              d
+        //
+        const auto area_threshold = std::pow(
+            scaledThresholdByRoadClass(MAX_SLIPROAD_THRESHOLD, sliproad_data.road_classification),
+            2.);
+
+        if (!isValidSliproadArea(
+                area_threshold, intersection_node_id, next->node, sliproad_edge_target))
+        {
+            continue;
+        }
+
+        // Check all roads at `d` if one is connected to `c`, is so `bd` is Sliproad.
+        for (const auto &candidate_road : target_intersection)
+        {
+            const auto &candidate_data = node_based_graph.GetEdgeData(candidate_road.eid);
+
+            // Name mismatch: check roads at `c` and `d` for same name
+            const auto name_mismatch = [&](const NameID road_name_id) {
```

Comment:
are you considering `name_a` -> `name_b` announcement does not necessarily require `name_b -> name_a` announcement.
Checking for `EmptyNameID` is missing here and should probably be here.